spec/models/searchgov_url_spec.rb

```diff
@@ -381,6 +381,37 @@
       end
     end
 
+    context 'when the searchgov domain has js renderer enabled' do
+      let(:js_searchgov_url) { described_class.create!(url: 'https://www.medicare.gov/sign-up-change-plans') }
```

Comment:
I'm still mulling through the larger testing philosophy for these changes, but didn't want to hold off my review into the weekend because I'm still thinking.

As this stands, these tests will actually send requests to these urls each time the tests are run, which is something we want to avoid.  See https://github.com/GSA/search-gov/wiki/Editing-Recording-and-Re-recording-API-calls-with-VCR-(WIP)

This may also be overkill for what we're actually trying to test here -- that the JsFetcher module receives a fetch request when SearchgovUrl#fetch is issued on a SearchgovDomain that has js_renderer set to true.  Checking the content of the response here doesn't quite feel right.

The inner logic of what is returned by JsFetcher probably belongs in a JsFetcher spec in `spec/lib`.  Happy to chat more on this while I continue to mull it over next week.

Reply:
So, after quick tinkering, something more like this might do the trick (this should still be cleaned up to reduce redundancy):

```
context 'when the searchgov domain has js renderer enabled' do
      before do
        allow(searchgov_url).to receive(:searchgov_domain).and_return(domain_with_js_renderer)
        allow(JsFetcher).to receive(:fetch)
      end

      let(:url) { 'https://www.medicare.gov/sign-up-change-plans' }
      let(:domain_with_js_renderer) do
        instance_double(SearchgovDomain, check_status: '200 OK', available?: true, js_renderer: true)
      end

      it 'fetches page with javascript fetcher' do
        fetch
        expect(JsFetcher).to have_received(:fetch).with(url)
      end
    end

    context 'when the searchgov domain has js renderer disabled' do
      before do
        allow(searchgov_url).to receive(:searchgov_domain).and_return(domain_without_js_renderer)
        allow(JsFetcher).to receive(:fetch)
      end

      let(:url) { 'https://www.medicare.gov/sign-up-change-plans' }
      let(:domain_without_js_renderer) do
        instance_double(SearchgovDomain, check_status: '200 OK', available?: true, js_renderer: false)
      end

      it 'fetches page without javascript fetcher' do
        fetch
        expect(JsFetcher).not_to have_received(:fetch).with(url)
      end
    end
```